### PR TITLE
Add initialization, use unittest framework

### PR DESCRIPTION
**Pull request by WimVanRoy**

### CODE REVIEW

**Comment by WimVanRoy** [comment]
For some reason, the checks in test_ocp were not applied neither where they working. I disabled the test.

File: nosnoc/nosnoc.py [hunk 1]
```diff
@@ -1204,13 +1204,25 @@ def solve(self) -> dict:
...
         """
         prob = self.problem
         dims = prob.model.dims
-        if field == 'x':
+        if field == 'x0':
             prob.model.x0 = value
+        elif field == 'x':
+            if value.shape[0] == self.opts.N_stages + 1:
+                # Shape is equal to the number of stages
+                self.set("x0", value[1, :])
```
Comment (FreyJo) [inline nosnoc/nosnoc.py:1217]: I would not make x0 part of this, especially, now that it isn't an optimization variable.

File: nosnoc/nosnoc.py [hunk 2]
```diff
@@ -1204,13 +1204,25 @@ def solve(self) -> dict:
...
+                for i in range(self.opts.N_stages):
+                    for indx in prob.ind_x[i+1]:
```
[the remark this answers is not in the document]
Comment (FreyJo) [reply nosnoc/nosnoc.py:1219]: So it seems that actually just the first few x values are initalized.

**Review by FreyJo — COMMENTED** [review]
Thanks @WimVanRoy !
I really like the unit tests.
2 Open points remain here for me:

- x initialization: see comments. Do you want to do this?
- test_ocp : check what doesnt work. I could do it next week.

File: nosnoc/nosnoc.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (FreyJo) [reply nosnoc/nosnoc.py]: Fixed in https://github.com/FreyJo/nosnoc_py/pull/20
thanks for rebasing!

**Comment by WimVanRoy** [comment]
Since the proper output of tests is useful, I added parametrized if you want to test a large set of the same problem. In this case, each combination will be one test that you can also disable if you would want to...

**Review by FreyJo — APPROVED** [review]
I pushed some small changes, mainly removing some test cases in `test_cop` that I added unintentionally, option combinations which aren't recommended and actually raise warnings.
I really like the new test and would merge as is!
Thanks for the contribution!

**Pull request merged**